### PR TITLE
[Enhancement] #431 catch enumerable errors to fields in census

### LINKED ISSUE #431: [Enhancement] Add enumerable display to available fields in census import

### PR DESCRIPTION
**Pull request by JavaLavaMT**
closes #431 

related associatedemployers/igneous#107

### CODE REVIEW

File: app/components/census/table-cell.js [hunk 1]
```diff
@@ -19,6 +19,17 @@ const CensusTableCellComponent = Component.extend(addEdit, ajaxStatus, {
     return field.isRelationship && column && !potentialDataForCell ? field.path : null;
   }),
 
+  hasEnumInvalidation: computed('availableFields.[]', 'column', 'columnIndex', function () {
```
Comment (James1x0) [inline app/components/census/table-cell.js]: I believe you are actually only using guessedField.enums now... Please change these dependent keys to just `'guessField.enums.[]'`

This is assuming you're not accounting for the following error... **What if the user entered a correct value?**

File: app/components/census/table-cell.js [hunk 2]
```diff
@@ -19,6 +19,17 @@ const CensusTableCellComponent = Component.extend(addEdit, ajaxStatus, {
...
+    if (this.guessedField.enums) {
+      let enums = this.guessedField.enums;
+      enums.push(' or leave it blank ');
+      enums = enums.filter(Boolean).join(', ');
```
Comment (James1x0) [inline app/components/census/table-cell.js]: Pushing is an altering move. Also with proper chaining, we can make this almost a one liner.

```
const enums = this.guessedField.enums,
    enumStr = [ ...enums, enums.indexOf(null) > -1 ? 'or leave this field blank' : null ].filter(Boolean).join(', ');
```

File: app/components/census/table-cell.js [hunk 3]
```diff
@@ -19,6 +19,17 @@ const CensusTableCellComponent = Component.extend(addEdit, ajaxStatus, {
...
+
+      let enumError = `Please use one of these: ${enums}`;
+      return enums  ? enumError : false;
```
Comment (James1x0) [inline app/components/census/table-cell.js]: I'd rather these two lines be a one liner (with proper spacing), and should be anyways if taking into account the error stated above.

File: app/controllers/account/employees/add/census/review.js [hunk 4]
```diff
@@ -56,10 +56,10 @@ export default Controller.extend(ajaxStatus, {
   }),
 
   availableFields: computed('model.availableFields.[]', function () {
-    return (this.get('model.availableFields') || []).map(({ path, format }) => ({
+    return (this.get('model.availableFields') || []).map(({ path, format, enums }) => ({
       path,
-      label:          `${this.convertPathToLabel(path)} - ${format}`,
-      isRelationship: format === 'lookup or id' ? true : false
```
Comment (James1x0) [inline app/controllers/account/employees/add/census/review.js]: Any reason why you have removed isRelationship here?? 🤨

File: app/templates/components/census/table-cell.hbs [hunk 5]
```diff
@@ -6,5 +6,10 @@
     </a>
   {{/ui-popup}}
 {{else}}
+  {{#if hasEnumInvalidation}}
+    {{#ui-popup tagName='span' content=hasEnumInvalidation position="right center"}}
+      <i class="exclamation circle icon text-red"></i>
+    {{/ui-popup}}
+  {{/if}}
```
Comment (James1x0) [inline app/templates/components/census/table-cell.hbs]: This looks good.

File: app/components/census/table-cell.js (hunk 2 above)
Comment (JavaLavaMT) [reply app/components/census/table-cell.js]: ^ I like this

File: app/controllers/account/employees/add/census/review.js (hunk 4 above)
Comment (JavaLavaMT) [reply app/controllers/account/employees/add/census/review.js]: no clue???

File: app/templates/components/census/table-cell.hbs (hunk 5 above)
Comment (JavaLavaMT) [reply app/templates/components/census/table-cell.hbs]: everyone gets one

File: app/controllers/account/employees/add/census/review.js (hunk 4 above)
Comment (James1x0) [reply app/controllers/account/employees/add/census/review.js]: @JavaLavaMT Lets add it back.

**Comment by James1x0** [comment]
@JavaLavaMT There are conflicting files here.

**Pull request merged**